### PR TITLE
[SYCL][NFC] Refactor [[intel::reqd_sub_group_size()]] attribute tests

### PR DESCRIPTION
**Pull request by smanna12**
This patch updates tests for [[intel::reqd_sub_group_size()]] attribute:

1. separates AST and diagnostics to make it easier to read/follow-up.
2. removes duplicate test cases.
3. adds missing tests.
4. no compiler changes.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/test/SemaSYCL/reqd-sub-group-size-sycl-kernel.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -fsyntax-only -sycl-std=2017 -Wno-sycl-2017-compat -verify -pedantic %s
+
+// The test checks functionality of [[intel::reqd_sub_group_size()]] attribute on SYCL kernel.
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/SemaSYCL/reqd-sub-group-size-sycl-kernel.cpp:3]: previously both diagnostics on sycl kernels and ast was in same file. So i decided to separate them.

**Review by elizabethandrews — APPROVED** [review]

File: clang/test/SemaSYCL/reqd-sub-group-size-ast.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+class KernelFunctor {
+public:
+  [[intel::reqd_sub_group_size(SIZE)]] void operator()() const {}
+};
+
+// Test that checks template parameter support on function.
+// CHECK: FunctionTemplateDecl {{.*}} func
+// CHECK: FunctionDecl {{.*}} func 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK_NEXT: IntelReqdSubGroupSizeAttr {{.*}} reqd_sub_group_size
```
Comment (pcolberg) [inline clang/test/SemaSYCL/reqd-sub-group-size-ast.cpp:35]: @smanna12 @pvchupin Is this line causing the `check-clang` failure in https://github.com/intel/llvm/pull/6220#issuecomment-1147803208?

**Comment by pvchupin** [comment]
Reverted after a chat with @smanna12

File: clang/test/SemaSYCL/reqd-sub-group-size-ast.cpp (hunk 2 above)
Comment (smanna12) [reply clang/test/SemaSYCL/reqd-sub-group-size-ast.cpp:35]: Yes, this is due to recent community change.  Thanks @pvchupin for reverting  the patch here: https://github.com/intel/llvm/pull/6258. 

Tagging @bader

**Pull request merged**